`azurerm_databricks_workspace` - Crash on Read

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_databricks_workspace`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

### Debug Output


### Panic Output

Output here: https://github.com/hashicorp/terraform-provider-azurerm/issues/12874#issuecomment-899664271

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x56dfe46]

goroutine 5198 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/databricks.resourceDatabricksWorkspaceRead(0xc0027b7480, 0x640dd80, 0xc001279c00, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/databricks/databricks_workspace_resource.go:567 +0x9e6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000497ce0, 0x73afae8, 0xc001bcb240, 0xc0027b7480, 0x640dd80, 0xc001279c00, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x1ee
```


### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

--- a/internal/services/databricks/databricks_workspace_resource.go
+++ b/internal/services/databricks/databricks_workspace_resource.go
@@ -548,23 +548,23 @@ func resourceDatabricksWorkspaceRead(d *pluginsdk.ResourceData, meta interface{}
 		if props.PublicNetworkAccess == databricks.PublicNetworkAccessDisabled {
 			d.Set("network_security_group_rules_required", string(props.RequiredNsgRules))
 		}
-		var cmkEnabled, infraEnabled *bool
+		var cmkEnabled, infraEnabled bool
 
 		if props.Parameters != nil {
-			if props.Parameters.PrepareEncryption != nil {
-				cmkEnabled = props.Parameters.PrepareEncryption.Value
-				d.Set("customer_managed_key_enabled", &props.Parameters.PrepareEncryption.Value)
+			if props.Parameters.PrepareEncryption != nil && props.Parameters.PrepareEncryption.Value != nil {
+				cmkEnabled = *props.Parameters.PrepareEncryption.Value
 			}
+			d.Set("customer_managed_key_enabled", cmkEnabled)
 
-			if props.Parameters.RequireInfrastructureEncryption != nil {
-				infraEnabled = props.Parameters.RequireInfrastructureEncryption.Value
-				d.Set("infrastructure_encryption_enabled", &props.Parameters.RequireInfrastructureEncryption.Value)
+			if props.Parameters.RequireInfrastructureEncryption != nil && props.Parameters.RequireInfrastructureEncryption.Value != nil {
+				infraEnabled = *props.Parameters.RequireInfrastructureEncryption.Value
 			}
+			d.Set("infrastructure_encryption_enabled", infraEnabled)
 
 			// The subnet associations only exist in the statefile, so we need to do a Get before we Set
 			// with what has come back from the Azure response...
 			customParamsRaw := d.Get("custom_parameters").([]interface{})
-			_, pubSubAssoc, priSubAssoc := expandWorkspaceCustomParameters(customParamsRaw, *cmkEnabled, *infraEnabled, "", "")
+			_, pubSubAssoc, priSubAssoc := expandWorkspaceCustomParameters(customParamsRaw, cmkEnabled, infraEnabled, "", "")
 
 			custom, backendPoolReadId := flattenWorkspaceCustomParameters(props.Parameters, pubSubAssoc, priSubAssoc)
 			if err := d.Set("custom_parameters", custom); err != nil {

